From 760ba6a87e9b7fe3012ce2ecd37757a649640262 Mon Sep 17 00:00:00 2001
From: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
Date: Thu, 7 Jul 2022 19:40:00 +0530
Subject: [PATCH] ath12k: fix 6g station association failure

In 6G STA, setting PEER_AUTH flag before completion of 4 way handshake causing
fw to drop M4 packet(EAPOL) and it results in association failure.

In general, PEER_AUTH flag should not to be set before completion of 4 way
handshake for security mode and clear the PEER_AUTH flag based on rsnie
presence to know security or open mode.

For 6G, clearing the PEER_AUTH bit irrespective of rsnie presense because it
does not support open mode.

Tested on: WLAN.WBE.1.0-02221-QCAHKSWPL_SILICONZ-1

Signed-off-by: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 510b62c..38c0f8c 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -1547,7 +1547,7 @@ static void ath12k_peer_assoc_h_crypto(struct ath12k *ar,
 	}
 
 	/* FIXME: base on RSN IE/WPA IE is a correct idea? */
-	if (rsnie || wpaie) {
+	if (ar->supports_6ghz || rsnie || wpaie) {
 		ath12k_dbg(ar->ab, ATH12K_DBG_WMI,
 			   "%s: rsn ie found\n", __func__);
 		arg->need_ptk_4_way = true;
-- 
2.7.4

